src/channel-details.js

```diff
@@ -169,6 +170,12 @@ class ChannelDetails extends ChannelDetailsBase {
         const reversed = []
         for (let i = msgs.length - 1; i >= 0; --i) {
           const msg = msgs[i]
+          if (bans.includes(msg.key)) {
+            if (msg.value && msg.value.content&& msg.value.content.text) {
+              msg.value.content.text = "this peer's content has been hidden"
```

Comment:
This seems like it would be quite easy for spammers to still disrupt the user experience here. For opportunists,the name could also be offensive, and be used to send messages by changing their handles often. Do you plan to implement a user experience in both desktop and cli for also hiding these messages? Ideally, it would be by default turned off. This seems like too risky to include now. It's not clear either what the extra benefit is of including 'this peers content has been hidden' messages. Did this come from user research?

Reply:
this is just a really shitty hack before we manage to get bans propagating into the actual message views. i noted that in the initial description of the PR

definitely not anything that will be final in anyway (as you note, showing the user name is an abuse vector as well. that's part of why the `banned` property is being set as well, but i figured this is as good as i could accomplish in the current PR so that we could get started testing the mod views and seeing changes in clients, without needing to PR them as well :)

Reply:
@noffle +1 go for it :)

Reply:
@cblgh Clarifying: are you saying you don't have spoons for this & are inviting me to do this, or are you saying that you'll do it?

Reply:
@noffle yup, sorry for being unclear! my spoon drawer is completely empty atm

as i wrote in the dev cabal (i think?), i have been focusing on my thesis mainly this week (submitting the first draft in 10 hours:^). i created this PR to add the changes i'd suggested in [substack's PR](https://github.com/cabal-club/cabal-client/pull/38), but after that i left it open so that anyone else could continue working on it; it could have been communicated a lot more clearly

thank you for clarifying!